04.ls/ls.rb

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+COLUMN_COUNT = 3
+
+def main
+  files = Dir.glob('*').sort_by(&:downcase)
+  row_count = files.count.ceildiv(COLUMN_COUNT)
+  columns = split_into_columns(files, COLUMN_COUNT, row_count)
+  output(columns, row_count)
+end
+
+def split_into_columns(files, column_count, row_count)
+  column_group = Array.new(column_count) { [] }
+  files.each_with_index do |file, index|
+    col_index = index / row_count
+    column_group[col_index] << file
+  end
+  column_group
+end
+
+def output(columns, row_count)
+  column_widths = columns.map { |words| words.map(&:size).max }
+  row_count.times do |row_idx|
+    columns.each_with_index do |col, col_idx|
+      filename = col[row_idx]
+      print "#{filename.ljust(column_widths[col_idx])}  " if filename
+    end
+    print "\n"
```

Comment:
参考情報）ちょっと高等テクニックなので今回はこのままでもいいですが、こんな書き方もできます。

```suggestion
    cols = columns.filter_map.with_index do |col, col_idx|
      filename = col[row_idx]
      filename.ljust(column_widths[col_idx]) if filename
    end
    puts cols.join('  ')
```

この場合のメリットは3列目の行末に無駄なスペースが発生しないことです。
以下の画像の赤色の部分が行末の無駄なスペースです。
（上がKimmyさん、下が僕が改善したコード）

<img width="357" alt="Screenshot 2025-06-06 at 10 03 43" src="https://github.com/user-attachments/assets/2da04807-6b8e-4510-97cd-d54b18e42d80" />